src/compiler/checker.ts

```diff
@@ -3328,25 +3329,62 @@ namespace ts {
             }
         }
 
-        // Returns true if the interface given by the symbol is free of "this" references. Specifically, the result is
-        // true if the interface itself contains no references to "this" in its body, if all base types are interfaces,
-        // and if none of the base interfaces have a "this" type.
-        function isIndependentInterface(symbol: Symbol): boolean {
+        function isIndependentBaseInterfaceList(nodes: NodeArray<ExpressionWithTypeArguments>): boolean {
+            if (!nodes) {
+                return true;
+            }
+            for (const node of nodes) {
+                if (!isIndependentHeritageClauseElement(node, /*classSymbol*/ undefined)) {

```

Comment:
Might be worth merging in Ron's `every` from the transforms branch and making `classSymbol` optional so you can just say

```ts
return !nodes || !every(nodes, isIndependentHeritageClauseElement)
```